packages/nimble-components/src/select/index.ts

```diff
@@ -785,17 +844,28 @@ export class Select extends FormAssociatedSelect implements ErrorPattern {
         const filter = this.filter.toLowerCase();
 
         if (filter) {
-            this.filteredOptions = this._options.filter(option => {
-                return diacriticInsensitiveStringNormalizer(
-                    option.text
-                ).includes(diacriticInsensitiveStringNormalizer(filter));
+            this.filteredOptions = this.options.filter(option => {
+                return (
+                    !option.hidden
+                    && diacriticInsensitiveStringNormalizer(option.text).includes(
+                        diacriticInsensitiveStringNormalizer(filter)
```

Comment:
`diacriticInsensitiveStringNormalizer(filter)` can we run that once outside the `filter` loop?